src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1589,6 +1503,32 @@ classifyWith f (Fold step1 initial1 extract1) =
 classify :: (Monad m, Ord k) => Fold m a b -> Fold m (k, a) (Map k b)
 classify fld = classifyWith fst (lmap snd fld)
 
+-- | Fold a stream of key value pairs using a map of specific folds for each
+-- key into a map from keys to the results of fold outputs of the corresponding
+-- values.
+--
+-- >>> import qualified Data.Map
+-- >>> :{
+--  let table = Data.Map.fromList [("SUM", Fold.lmap snd Fold.sum), ("PRODUCT", Fold.lmap snd Fold.product)]
+--      input = Stream.fromList [("SUM",1),("PRODUCT",2),("SUM",3),("PRODUCT",4)]
+--   in Stream.fold (Fold.demux table Fold.sum) input
+-- :}
+-- fromList [("PRODUCT",8),("SUM",4)]
+--
+-- > demux kv fld = demuxDefaultWith fst kv (lmap snd fld)
+--
+-- /Pre-release/
+{-# INLINE demux #-}
+demux ::  (Monad m, Ord k)
+    => Map k (Fold m (k, a) b) -> Fold m a b -> Fold m (k, a) (Map k b)
+demux kv fld = demuxDefaultWith fst kv (lmap snd fld)
+
+-- /Pre-release/
+{-# INLINE demux_ #-}
+demux_ :: (Monad m, Ord k)
+    => Map k (Fold m (k, a) ()) -> Fold m (k, a) (Map k ())
+demux_ kv = demux kv drain
+
```

Comment:
Why are these moved after classify? We can put these right after demuxDefaultWith.

Reply:
We can remove `demux_`.